Prog/AddNew/TOPA_data_to_DC_data.sas

```diff
@@ -0,0 +1,155 @@
+/**************************************************************************
+ Program:  DC_data_to_TOPA_data.sas
+ Library:  PresCat
+ Project:  Urban-Greater DC
+ Author:   Elizabeth Burton
+ Created:  9/29/2022
+ Version:  SAS 9.4
+ Environment:  Local Windows session (desktop)
+ 
+ Description:  Match TOPA database from CNHED to DCData data sets to add characteristics. 
+ 
+ Modifications:
+**************************************************************************/
+
+%include "\\sas1\DCdata\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( PresCat )
+%DCData_lib( DHCD )
+%DCData_lib( MAR )
+%DCData_lib( RealProp )
+
+** Download and read TOPA dataset into SAS dataset**;
+%let dsname="&_dcdata_r_path\PresCat\Raw\TOPA\TOPA_DOPA_5+.csv";
+filename fixed temp;
+/** Remove carriage return and line feed characters within quoted strings **/
+/*'0D'x is the hexadecimal representation of CR and
+/*'0A'x is the hexadecimal representation of LF.*/
+/* Replace carriage return and linefeed characters inside */
+/* double quotes with a specified character.  */
+/* CR/LFs not in double quotes will not be replaced. */
+%let repA=' || '; /* replacement character LF */
+%let repD=' || '; /* replacement character CR */
+ data _null_;
+ /* RECFM=N reads the file in binary format. The file consists */
+ /* of a stream of bytes with no record boundaries. SHAREBUFFERS */
+ /* specifies that the FILE statement and the INFILE statement */
+ /* share the same buffer. */
+ infile &dsname recfm=n sharebuffers;
+ file fixed recfm=n;
+ /* OPEN is a flag variable used to determine if the CR/LF is within */
+ /* double quotes or not. Retain this value. */
+ retain open 0;
+ input a $char1.;
+ /* If the character is a double quote, set OPEN to its opposite value. */
+ if a = '"' then open = ^(open);
+ /* If the CR or LF is after an open double quote, replace the byte with */
+ /* the appropriate value. */
+ if open then do;
+ if a = '0D'x then put &repD;
+ else if a = '0A'x then put &repA;
+ else put a $char1.;
+ end;
+ else put a $char1.;
+run;
+
+proc import datafile=fixed
+            dbms=csv
+            out=TOPA_database1
+            replace;
+		
+     getnames=yes;
+	 guessingrows=max;
+run;
+
+proc print data=TOPA_database1 (obs=5); 
+run; 
+
+data TOPA_database;
+    set TOPA_database1;
+    CASD_date = input(CASD_Report_week_ending_date, MMDDYY10.);
+    format CASD_date MMDDYY10.;
+    drop CASD_Report_week_ending_date;
```

Comment:
I wouldn't drop the original vars as it can be helpful to see them if we need to check the date coding.

Reply:
It would also be helpful to generate a list of notices with missing dates. Would you please add a Proc Print to the program notices that are missing sale date and CASD report date that we can share back with CNHED? We don't want to  include all variables--just key ones that will help them identify those notices.